pkgs/build-support/setup-hooks/auto-patchelf.py

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env python3
+
+from collections import defaultdict
+from contextlib import contextmanager
+from dataclasses import dataclass
+from elftools.common.exceptions import ELFError
+from elftools.elf.dynamic import DynamicSection
+from elftools.elf.elffile import ELFFile
+from elftools.elf.enums import ENUM_E_TYPE, ENUM_EI_OSABI
+from itertools import chain
+from pathlib import Path, PurePath
+from typing import Tuple
+
+import argparse
+import os
+import pprint
+import subprocess
+import sys
+
+
+
+@contextmanager
+def open_elf(path):
```

Comment:
```suggestion
def open_elf(path: str) -> Iterator[IO[byte]]:
```

Reply:
The `path` argument actually has type `Path`